src/checkbox/src/Checkbox.js

```diff
@@ -35,6 +34,42 @@ MinusIcon.propTypes = {
 
 const noop = () => {}
 
+const pseudoSelectors = {
+  _base: '& + div',
+  _disabled: '&[disabled] + div',
+  _hover: '&:not([disabled]):hover + div',
+  _focus: '&:not([disabled]):focus + div',
+  _active: '&:not([disabled]):active + div',
+  _checked: '&:checked + div, &[type=checkbox]:indeterminate + div',
+  _checkedHover:
+    '&:not([disabled]):checked:hover + div, &[type=checkbox]:not([disabled]):indeterminate:hover + div',
+  _checkedActive:
+    '&:not([disabled]):checked:active + div, &[type=checkbox]:not([disabled]):indeterminate:active + div',
+  _checkedDisabled:
+    '&[disabled]:checked + div, &[type=checkbox][disabled]:indeterminate + div'
```

Comment:
Not for this PR but I would love to really understand why we need these combinations (presumably to preserve specificity). They feel non-standard since they are highly tailored to the checkbox component.

Reply:
I'm with you here - I think it's worth re-visiting a lot of the psuedoselectors throughout EG in general.

Reply:
+1 on revisiting the selectors